pandas/core/arrays/sparse/dtype.py

```diff
@@ -127,7 +127,15 @@ def __eq__(self, other: Any) -> bool:
                     or isinstance(other.fill_value, type(self.fill_value))
                 )
             else:
-                fill_value = self.fill_value == other.fill_value
+                with warnings.catch_warnings():
```

Comment:
Do we know that this isn't something that we need to fix?

Reply:
this is issued from numpy, i think numpy/numpy#22707 fixes it